benchmarks/entropy_adiabat/plugins/entropy_advection.cc

```diff
@@ -181,17 +209,19 @@ namespace aspect
           const double u_grad_field = u * (scratch.old_field_grads[q] +
                                            scratch.old_old_field_grads[q]) / 2;
 
-          const double density       = scratch.material_model_outputs.densities[q];
-          const double entropy_conductivity  = scratch.material_model_outputs.thermal_conductivities[q] *
-                                               scratch.material_model_inputs.temperature[q] /
-                                               scratch.material_model_outputs.specific_heat[q];
-          const double k_Delta_field = entropy_conductivity * (scratch.old_field_laplacians[q] +
-                                                               scratch.old_old_field_laplacians[q]) / 2;
+          const double density      = scratch.material_model_outputs.densities[q];
+          const double gamma        = scratch.heating_model_outputs.heating_source_terms[q];
+
+          // Because we solve the diffusion equation for the temperature before the advection equation, we can use
+          // the current and old temperature here, together with the current time step.
+          const double diffusion_term = (this->get_timestep() == 0.0) ? 0.0
+                                        :
+                                        (scratch.material_model_inputs.temperature[q] - old_temperatures[q]) / this->get_timestep()
```

Comment:
I am slightly concerned about `scratch.material_model_inputs.temperature[q]` here. all other properties here are computed at time `old_... + old_old_... /2` and I think we would need the same for the diffusion term? I.e. `scratch.old_temperature_values[q] - scratch.old_old_temperature_values[q]`. That would also align it with how `dField_dt` is computed.

Reply:
My reasoning for this was that the diffusion term is only first order, so the current minus the old temperature is what is actually being used in the diffusion equation. This is also exactly what is being used in the actual assembly.

One way to see it is that now this term is explicit instead of implicit, so the values we use here can be the exact same values being used in the assembly. So this now works the same way as a term like gamma.